packages/angular/cli/upgrade/version.ts

```diff
@@ -110,67 +109,4 @@ export class Version {
     }
   }
 
-  static assertTypescriptVersion(projectRoot: string) {
-    if (!isWarningEnabled('typescriptMismatch')) {
```

Comment:
We should also deprecate, or wholly remove, this option. There are many files that reference it, including tests.

Reply:
Cleaned up the option as well.